lib/libc/minimal/Kconfig

```diff
@@ -14,31 +14,43 @@ config MINIMAL_LIBC_NON_REENTRANT_FUNCTIONS
 	  option may require an additional memory protection region.
 
 config MINIMAL_LIBC_MALLOC
-	bool "Minimal libc malloc implementation"
+	bool "[DEPRECATED] Minimal libc malloc implementation"
 	default y
+	select COMMON_LIBC_MALLOC
 	help
+	  [DEPRECATED] Use COMMON_LIBC_MALLOC
+
 	  Enable the minimal libc's implementation of malloc, free, and realloc.
 	  Disable if you wish to provide your own implementations of these functions.
 
 config MINIMAL_LIBC_MALLOC_ARENA_SIZE
-	int "Size of the minimal libc malloc arena"
+	int "[DEPRECATED] Size of the minimal libc malloc arena"
 	default 0
-	depends on MINIMAL_LIBC_MALLOC
+	depends on COMMON_LIBC_MALLOC
 	help
+	  [DEPRECATED] Use COMMON_LIBC_MALLOC_ARENA_SIZE
+
+	  If set to zero, then the value of COMMON_LIBC_MALLOC_ARENA_SIZE
+	  will be used.
+
 	  Indicate the size in bytes of the memory arena used for
 	  minimal libc's malloc() implementation.
 
 config MINIMAL_LIBC_CALLOC
-	bool "Minimal libc trivial calloc implementation"
+	bool "[DEPRECATED] Minimal libc trivial calloc implementation"
 	default y
```

Comment:
To deprecate these, they must not be selected by default, and they must select DEPRECATED

Reply:
I don't know how to make that compatible with existing uses where this item is enabled by default.

Reply:
Isn't `COMMON_LIBC_CALLOC` a direct replacement for this, i.e. users would notice no difference by switching? If so, these can be unselected by default and the new ones selected.

Reply:
But that doesn't seem aligned with the Zephyr deprecation guidelines which ask that existing uses remain supported for two releases -- existing configurations selecting `CONFIG_MINIMAL_LIBC_CALLOC=n` should have calloc disabled, even if they don't include `CONFIG_COMMON_LIBC_CALLOC=n`, right? 

Configurations without any setting for `CONFIG_MINIMAL_LIBC_CALLOC` or `CONFIG_COMMON_LIBC_CALLOC` should have calloc enabled. Suggestions on how to achieve that while `MINIMAL_LIBC_CALLOC` doesn't use `default y` would be most appreciated.

Reply:
People can still use the existing method, they will just get a deprecation warning when they configure the build, the default configuration does not enable any deprecated options, it would move to the new options. If the existing way is `default y` for the old Kconfig, then use the same for the new or `default y if !MINIMAL_LIBC_MALLOC`, and do not `default y` for the old options

Reply:
These minimal libc options are a bit unusual -- they would only be specifically configured as 'n', never 'y'. If the default is set to 'n', how do we detect an explicit setting of 'n' so that the existing configuration would continue to work?

Reply:
The new system should be a drop in replacement for the old, if people want to use the old system, it still needs to be possible, but I would say allowing them to de-select the new options and select the old options is a good enough compromise, that will give them the deprecated option notice, and then after 2 releases the old options can be removed entirely. The new system can be selected by default (I thought the idea was to switch to picolibc as the default anyway?)

Reply:
I don't know how to meet your goals and still keep existing configurations working. In particular, how would an existing configuration which selects `CONFIG_MINIMAL_LIBC_CALLOC=n` but does not include `CONFIG_COMMON_LIBC_CALLOC=n` work? 

We must be able to distinguish configurations which set `CONFIG_MINIMAL_LIBC_CALLOC=n` from configurations which have no value at all, and I don't know how to do that without using a default value of `y`.

Reply:
As for moving to picolibc by default, this is part of that effort -- we want to switch picolibc configurations to use the system allocator instead of the picolibc malloc implementation. Having two separate allocators in the system isn't a good use of resources.

Reply:
@tejlmand for comment/review